projects/dotcms-js/src/lib/core/core-web.service.mock.ts

```diff
@@ -1,29 +1,103 @@
 import { Injectable } from '@angular/core';
 import { Observable, of } from 'rxjs';
-import { RequestOptionsArgs, Http, Request } from '@angular/http';
-import { map } from 'rxjs/operators';
+import { map, filter } from 'rxjs/operators';
 import { ResponseView } from './util/response-view';
+import {
+    HttpClient,
+    HttpRequest,
+    HttpEvent,
+    HttpEventType,
+    HttpResponse,
+    HttpParams,
+    HttpHeaders
+} from '@angular/common/http';
+import { RequestMethod, RequestOptionsArgs } from '@angular/http';
+import { RequestOptionsParams } from 'dotcms-js';
 
 @Injectable()
 export class CoreWebServiceMock {
+    constructor(private _http: HttpClient) {}
 
-    constructor(private _http: Http) {}
+    request(options: RequestOptionsArgs): Observable<any> {
+        const optionsArgs: RequestOptionsParams = {
+            params: new HttpParams()
+        };
 
-    request(options: any): Observable<any> {
-        return this._http.request(new Request(options));
+        if (options.params) {
+            Object.keys(options.params).forEach((key) => {
+                optionsArgs.params = optionsArgs.params.set(key, options.params[key]);
+            });
+        }
+
+        return this._http
+            .request(
+                new HttpRequest(
+                    RequestMethod[options.method],
+                    options.url,
+                    options.body,
+                    { params: optionsArgs.params }
+                )
+            )
+            .pipe(
+                filter(<T>(event: HttpEvent<T>) => event.type === HttpEventType.Response),
```

Comment:
What is this filter for?

Reply:
when using `HttpClient.request` the response received is a `HttpEvent`. 

The first response/event is `{type: 0}` which indicates `the request has been dispatched to the backend`, after that the next response will contain the actual content from the BackEnd

*We need to listen for `events` to get the entire request from the BE, otherwise if we listen only for the content then we'll miss the headers and other information from the request that is needed (for example in pagination)